src/internal/types.ts

```diff
@@ -218,3 +218,19 @@ type InternalIsUnion<T, U = T> = (
     ? true
     : Result
   : never; // Should never happen
+
+/**
+ * Extracts a type predicate from a type guard function for the first argument.
+ *
+ * @example
+ * type TypeGuardFn = (x: unknown) => x is string;
+ * type Result = GuardType<TypeGuardFn>; // `string`
+ */
+export type GuardType<T, Fallback = never> = T extends (
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any -- we don't care about arguments types here
+  x: any,
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any -- we don't care about arguments types here
+  ...rest: any
```

Comment:
**question:** is there a reason to use `any` here rather than `unknown`?

Reply:
strange! i think this is fine, but i was just curious...

Reply:
@cjquines 

https://stackoverflow.com/questions/62136028/why-dont-parametersfunc-extend-unknown-arrayunknown-in-ts-strict-mode

![image](https://github.com/remeda/remeda/assets/10157660/9ffe3684-9688-4f3a-8155-266611018141)

Reply:
It seems we can use `never` but it's probably the same as `any` in this case.